niCHARTPipelines/__main__.py

```diff
@@ -6,37 +6,281 @@
-
-    inDir = args.inDir
-    DLICVmdl = args.DLICVmdl
-    DLMUSEmdl = args.DLMUSEmdl
-    pipelineType = args.pipelineType
-    outDir = args.outDir
+    prog = "niCHARTPipelines"
+    description = "niCHART Data Preprocessing Pipelines"
+    usage = """
+    niCHARTPipelines v{VERSION}
+    ICV calculation, brain segmentation, and ROI extraction pipelines for 
+    structural MRI data.
+
+    required arguments:
+        [INDIR]         The filepath of the directory containing the input. The 
+        [-i, --indir]   input can be a single .nii.gz (or .nii) file or a  
+                        directory containing .nii.gz files (or .nii files). 
+
+        [OUTDIR]        The filepath of the directory where the output will be
+        [-o, --outdir]  saved.
+
+        [PIPELINETYPE]  Specify type of pipeline[structural, dti, fmri]. 
+        [-p,            Currently only structural pipeline is supported.
+        --pipelinetype]
+
+        [SCANID]        Scan ID of the subject, if only one scan is given. If
+        [-s, --scanID]  multiple scans are given, this argument will be 
+                        ignored, and the scan ID will be extracted from the
+                        filename of the input file(s).
+
+        [DERIVED_ROI_MAPPINGS_FILE]     The filepath of the derived MUSE ROI 
+        [--derived_ROI_mappings_file]   mappings file.
+
+        [MUSE_ROI_MAPPINGS_FILE]    The filepath of the MUSE ROI mappings file.
+        [--MUSE_ROI_mappings_file]
+    
+    optional arguments: 
+        [DLICVMDL]      The filepath of the DLICV model will be. In case the
+        [--DLICVmdl]    model to be used is an nnUNet model, the filepath of
+                        the model's parent directory should be given. Example: 
+                        /path/to/nnUNetTrainedModels/nnUNet/
+        
+        [DLMUSEMDL]     The filepath of the DLMUSE model will be. In case the
+        [--DLMUSEmdl]   model to be used is an nnUNet model, the filepath of
+                        the model's parent directory should be given. Example:
+                        /path/to/nnUNetTrainedModels/nnUNet/
+
+        [NNUNET_RAW_DATA_BASE]   The filepath of the base directory where the 
+        [--nnUNet_raw_data_base] raw data of are saved.  This argument is only 
+                                 required if the DLICVMDL and DLMUSEMDL 
+                                 arguments are corresponding to a  nnUNet model 
+                                 (v1 needs this currently).
+
+        [NNUNET_PREPROCESSED]   The filepath of the directory where the 
+        [--nnUNet_preprocessed] intermediate preprocessed data are saved. This
+                                argument is only required if the DLICVMDL and
+                                DLMUSEMDL arguments are corresponding to a
+                                nnUNet model (v1 needs this currently).
+
+        [RESULTS_FOLDER]        THIS IS ONLY NEEDED IF BOTH DLICV AND DLMUSE 
```

Comment:
Better to rename this to 'MODEL_FOLDER'. Internally it can be used to set the os.environ['RESULTS_FOLDER'] env variable.

Reply:
Understood, agreed.